src/redisai.c

```diff
@@ -426,27 +418,67 @@ int RedisAI_Set_Graph_RedisCommand(RedisModuleCtx *ctx, RedisModuleString **argv
     return RedisModule_ReplyWithError(ctx, "ERR unsupported device");
   }
 
-  RAI_Graph *graph = NULL;
+  long long argidx = 4;
+
+  char **inputs = array_new(char*, argc - argidx);
```

Comment:
You can use a variable-length-array here with some bounds sanity checking, unless the expected argument size can really be a lot..

e.g:
```
size_t remaining = argc - argidx;
if (remaining > 1000) { return RedisModule_ReplyWithError(ctx, "Too many arguments"); }
const char *inputs[remaining];
const char *outputs[remaining];
```

Reply:
One question here: I'm using the `array` in utils so we can avoid passing the number of elements to the called functions. What's the actual advantage of avoiding `array` (it doesn't look like it has a lot of overhead)?

Reply:
It's not so much a matter of overhead as it is a matter of yet one more pointer to keep track of and ensure is freed.

Reply:
I see now, thanks.
The downside is making the signature for the called functions a bit more complicated (with the added ninputs, noutputs). I'm not sure about this one.